### ISSUE TITLE
Should not lint on function in function signature.

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#18]
## box.linters version

0.0.0.9006

## Source code to lint

```R
some_function <- function(x, y, func = NULL) {
  func(x, y)
}

some_function(2, 3, sum)
```

## Lint command

```R
lintr::lint(code, linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters))
```

## Lint result

```
<text>:3:3: warning: [box_usage_linter] Function not imported nor defined.
  func(x, y)
  ^~~~
```

## Expected result

Should not lint on columns `func()`.
